tests/test_migration_processing_guards.py

```
@@ -484,16 +348,17 @@ def test_process_inbox_restores_failed_sources_back_to_raw(temp_vault, monkeypat
 
     logger = ArticleLogger(temp_vault / "60-Logs" / "pipeline.jsonl")
     txn = ArticleTxn(temp_vault / "60-Logs" / "transactions")
-    processor = AutoArticleProcessor(temp_vault, logger, txn, skip_deep_dive=False)
-    processor.article_processor = SimpleNamespace(
-        generate_interpretation=lambda **kwargs: (_ for _ in ()).throw(RuntimeError("boom"))
-    )
+    processor = AutoArticleProcessor(temp_vault, logger, txn)
+
+    # Force the processor to fail by making image download throw —
+    # post-BL-029 this is the realistic failure surface.
+    def raising_process_file(self, file_path, backup=True):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `raising_process_file`

Add return type annotation: `NoReturn`

(ANN202)